test/integration/smart_proxy_test.rb

```diff
@@ -39,4 +43,47 @@ class SmartProxyIntegrationTest < ActionDispatch::IntegrationTest
     assert page.has_selector?('h3', :text => "DHCP")
     assert page.has_content? "Version"
   end
+
+  test "show page with conditional pagelets" do
+    Pagelets::Manager.add_pagelet("smart_proxies/show", :main_tabs, :name => "Visible", :partial => "../../test/static_fixtures/views/test", :onlyif => Proc.new { true })
+    Pagelets::Manager.add_pagelet("smart_proxies/show", :main_tabs, :name => "Hidden", :partial => "../../test/static_fixtures/views/test", :onlyif => Proc.new { false })
+    proxy = smart_proxies(:one)
+    visit smart_proxy_path(proxy)
+    assert page.has_link? "Visible"
+    refute page.has_link? "Hidden"
+  end
+
+  test "show page passes @smart_proxy into pagelets" do
```

Comment:
Can you move this and the following tests into unit tests for the helpers where possible?  Integration tests take seconds to run each, which might be OK to add one or two for basic functionality testing, but for these sort of combinations they really should be unit tested instead.